File: CHRISTIAN_PINEAPPLE/1-infra/stacks/AgentCoreFetchNewsAgentStack/createIamRole/createIamRole.ts
import { PolicyStatement, Role, ServicePrincipal } from "aws-cdk-lib/aws-iam";
import { Construct } from "constructs";
import {
  projectName,
  projectNamePascal,
  tenantName,
  tenantNamePascal,
} from "../../../constants";

export const createIamRole = (stack: Construct, envName: string) => {
  const logicalId = `${envName}${projectNamePascal}${tenantNamePascal}NewsAgentLambdaExecutionRole`;
  const name = `${envName}-${projectName}-${tenantName}---news-agent-execution-role`;

  const lambdaRole = new Role(stack, logicalId, {
    assumedBy: new ServicePrincipal("bedrock-agentcore.amazonaws.com"),
    roleName: name,
    description: `execution role for ${envName}-${projectName}-${tenantName}---news-agent`,
  });

  lambdaRole.addToPolicy(
    new PolicyStatement({
      actions: ["bedrock:*", "ecr:*", "logs:*"],
      resources: ["*"],
    })
  );

  return lambdaRole;
};

File: RISU_UNDERSTAND/infra/stacks/fetch_news_agent_stack/create_iam_role/create_iam_role.py
from aws_cdk import (
    aws_iam as iam,
)

def create_iam_role(scope):
    return iam.Role(
        scope, "ChristianBlueberryAgentRole",
        role_name="Christian-Blueberry-News-Agent-ExecutionRole",
        description="Execution role for Christian Blueberry News Aggregator Agent",
        assumed_by=iam.ServicePrincipal("bedrock-agentcore.amazonaws.com"),
        managed_policies=[
            iam.ManagedPolicy.from_aws_managed_policy_name(
                "BedrockAgentCoreFullAccess"
            )
        ],
        inline_policies={
            "ChristianBlueberryNewsAgentPolicy": iam.PolicyDocument(
                statements=[
                    iam.PolicyStatement(
                        sid="ECRImagePull",
                        effect=iam.Effect.ALLOW,
                        actions=[
                            "ecr:GetAuthorizationToken",
                            "ecr:BatchCheckLayerAvailability",
                            "ecr:GetDownloadUrlForLayer",
                            "ecr:BatchGetImage",
                        ],
                        resources=["*"],
                    ),
                    iam.PolicyStatement(
                        sid="BedrockModelInvocation",
                        effect=iam.Effect.ALLOW,
                        actions=[
                            "bedrock:InvokeModel",
                            "bedrock:InvokeModelWithResponseStream",
                        ],
                        resources=["*"],
                    ),
                    iam.PolicyStatement(
                        sid="CloudWatchLogs",
                        effect=iam.Effect.ALLOW,
                        actions=[
                            "logs:CreateLogGroup",
                            "logs:CreateLogStream",
                            "logs:PutLogEvents",
                            "logs:DescribeLogStreams",
                        ],
                        resources=["*"],
                    ),
                ]
            )
        },
    )